terraform/modules/daos_server/templates/daos_startup_script.tftpl

```diff
@@ -8,30 +8,47 @@ DAOS_MOUNT_DIR="/var/daos"
 fetch_attr()
 {
   local attr_name=$*
-  curl -s ${METADATA_URL}/${attr_name} -H "Metadata-Flavor: Google"
+  curl -s $${METADATA_URL}/$${attr_name} -H "Metadata-Flavor: Google"
 }
 
 echo "BEGIN: Setting up DAOS server"
 
 systemctl stop daos_server
 
 # Create server config files
-mkdir -p "${DAOS_CONFIG_DIR}"
-cd "${DAOS_CONFIG_DIR}"
+mkdir -p "$${DAOS_CONFIG_DIR}"
+# Create directory for engine logs and tmpfs mount point
+mkdir -p "$${DAOS_MOUNT_DIR}"
+cp /tmp/configure_daos.sh  $${DAOS_MOUNT_DIR}/configure_daos.sh
+cd "$${DAOS_CONFIG_DIR}"
+
```

Comment:
an alternative for the /tmp would be to render the configure_daos.sh and then fetch it from metadata the same way as the server, control and agent files.  Although overly relying on metadata is also a bad thing. If it does not need rendering, then maybe the image is the best place.